add storage for velocity controls to please TAF

timothyas opened the pull request:
## What changes does this PR introduce?
(Bug fix, feature, docs update, ...)

This is really a small fix. When initial uvel/vvel fields are in the control vector, i.e. 

`defined ALLOW_UVEL0_CONTROL && defined ALLOW_VVEL0_CONTROL`

TAF complains that it needs to recompute these fields within `packages_init_variables`. This probably isn't a big savings in compute time, but since this likely applies to all the ECCO, SOSE, ASTE, et al folks (at least), I thought I'd introduce the PR... 

## What is the current behaviour? 
(You can also link to an open issue here)

TAF gives a warning about some extra recomputations.

## What is the new behaviour 
(if this is a feature change)?

This is fixed.

## Does this PR introduce a breaking change? 
(What changes might users need to make in their application due to this PR?)


## Other information:


## Suggested addition to `tag-index`
(To avoid unnecessary merge conflicts, please don't update `tag-index`. One of the admins will do that when merging your pull request.)

jm-c commented:
@timothyas I took a quick look, and it seems it's the first time these 2 CPP options appear in main
model/src code. I seach for them in current verification exp. (as well as in verification_other) but did not find any example that defines any of these 2 options. They don't appear anywhere neither in any
pkg/${PKG}_OPTIONS.h file.
1) I would suggest to add/document these 2 options (set to #undef + 1 line of comment in
  one of the 2 files, either in:
   https://github.com/MITgcm/MITgcm/blob/master/pkg/ctrl/CTRL_OPTIONS.h
  (where we already have option for initial Theta & Salt controls)
  or in:
   https://github.com/MITgcm/MITgcm/blob/master/pkg/ecco/ECCO_OPTIONS.h
    although it seems less natural there.
2) not clear to me why you added these 2 storage dir only when both UVEL0 and VVEL0 options
  are defined. I would have thought that it would be safer to add these storage dir any time one of 
  the 2 is defined.

timothyas commented:
Thanks for the review @jm-c! 

1. Done, I added these to CTRL_OPTIONS.h since I agree, it seems more natural than ECCO_OPTIONS.h. I also added these to the documentation since it seems they aren't listed there either. 

2. I borrowed the same logic as what is in [ctrl_map_ini_genarr](https://github.com/MITgcm/MITgcm/blob/788c57c2b589e3e22327271c5a18ca45f27ae857/pkg/ctrl/ctrl_map_ini_genarr.F#L144) since this seems to be what is causing the TAF recomputation warning. I would guess that this is for LLC type grids, where the model uvel/vvel is not the usual definition.

The second issue brings up the fact that even if someone uses generic controls, they additionally have to define these macros to use initial velocity as a control variable. The same is true for bottom drag and geothermal flux (see ctrl_map_ini_genarr). Should I add a note about this in the documentation and/or CTRL_OPTIONS.h?

jm-c commented:
LGTM +1
Will try to merge this PR tomorrow.

timothyas commented:
Thanks @jm-c!
